src/network.js

```diff
@@ -1,51 +1,139 @@
+                                    'Content-Type': 'application/json'
+                                },
+                            })
+                            this.accessToken = tokenResponse.data.access_token;
+                            retried = true;
+                            executePromise();
+                        } catch(error) {
+                            reject(error);
+                        }
+                    } else {
+                        reject(error);
+                    }
+                });
+            }
+            executePromise();
+        })
+    }
+
+    #checkDisplayPromise() {
+        return new Promise((resolve, reject) => {
+            axios.get(this.url + '/rules', {
+                withCredentials: true,
+                headers: {
+                    'Access-Control-Allow-Origin': '*',
+                    'Content-Type': 'application/json',
+                    'Authorization': 'Bearer ' + this.accessToken
+                },
+                params: {
+                    featureUrl: this.featureUrl
+                },
+            }).then(result => {
+                resolve(result);
+            }).catch(error => {
+                reject(error);
+            });
+        });
+    }
+
+    #postCommentPromise(rating, comment) {
+        return new Promise(async (resolve, reject) => {
+            // Get the visitor identifier from fingerprint
+            const fp = await this.fpPromise;
+            const result = await fp.get();
+            const user_fingerprint = result.visitorId;
+
+            var data = {
+                rating: rating,
+                comment: comment,
+                feature_url: this.featureUrl,
+                user_id: user_fingerprint
+            };
+            axios.post(this.url + '/comments', data, {
+                withCredentials: true,
+                headers: {
+                    'Access-Control-Allow-Origin': '*',
+                    'Content-Type': 'application/json',
+                    'Authorization': 'Bearer ' + this.accessToken
+                }
+            }).then(result => {
+                resolve(result);
+            }).catch(error => {
+                reject(error);
+            });
+        });
     }
 
     /**
      * 
      * @returns {boolean} represents if the modal need to be displayed or not.
      */
     async initConfig() {
-        const response = await axios.get(this.url + '/rules', {
-            withCredentials: true,
-            headers: {
-                'Access-Control-Allow-Origin': '*',
-                'Content-Type': 'application/json'
-            },
-            params: {
-                featureUrl: this.featureUrl
-            },
+        const request = this.#sendRequestWithOAuth(this.#checkDisplayPromise.bind(this));
+        return request.then((response) => {
+            return response.data;
+        }).catch(error => {
+            console.log(error);
```

Comment:
You are using console.log to display errors. However, this does not interrupt the code execution and does not provide proper error handling. Instead, you should reject the promise with the error so that the caller can handle errors appropriately.
```suggestion
            reject(error);
```